### PR TITLE
Item 7563: SampleTypeDesigner update to add "Label Color" property to Sample Manager

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
There are expectations that there will be several sample types used in the Sample Manager and Freezer Manager apps. To distinguish them quickly in various parts of the UI, we are adding a "Label Color" property to the Sample Type so that it can be tagged with a color to be used in things like the Freezer box display and grid.

This PR adds the LabelColor field to the SampleSets table for persistence to the LK DB table. It also pulls in the updated @labkey/components package, to verify that the new field input is not currently shown in the LK sample type create/update designer UI.

#### Related Pull Requests
* https://github.com/LabKey/labkey-ui-components/pull/296
* https://github.com/LabKey/sampleManagement/pull/324
* https://github.com/LabKey/inventory/pull/54

#### Changes
* Update @labkey/components package version
* SQL upgrade script to add LabelColor column to exp.MaterialSource (and related Java class model/schema/table updates)

### CODE REVIEW

File: api/src/org/labkey/api/exp/api/SampleTypeService.java [hunk 1]
```diff
@@ -76,7 +76,7 @@ ExpSampleType createSampleType(Container container, User user, String name, Stri
 
     @NotNull
     ExpSampleType createSampleType(Container container, User user, String name, String description, List<GWTPropertyDescriptor> properties, List<GWTIndex> indices, int idCol1, int idCol2, int idCol3, int parentCol,
-                                   String nameExpression, @Nullable TemplateInfo templateInfo, Map<String, String> importAliases)
+                                   String nameExpression, @Nullable TemplateInfo templateInfo, Map<String, String> importAliases, String labelColor)
```
Comment (labkey-susanh) [inline api/src/org/labkey/api/exp/api/SampleTypeService.java]:
```suggestion
                                   String nameExpression, @Nullable TemplateInfo templateInfo, Map<String, String> importAliases, @Nullable String labelColor)

```

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java [hunk 2]
```diff
@@ -597,13 +597,13 @@ public ExpSampleTypeImpl createSampleType(Container c, User u, String name, Stri
...
         return createSampleType(c, u, name, description, properties, indices, idCol1, idCol2, idCol3,
-                parentCol, nameExpression, templateInfo, null);
+                parentCol, nameExpression, templateInfo, null, null);
     }
 
     @NotNull
     @Override
     public ExpSampleTypeImpl createSampleType(Container c, User u, String name, String description, List<GWTPropertyDescriptor> properties, List<GWTIndex> indices, int idCol1, int idCol2, int idCol3, int parentCol,
-                                              String nameExpression, @Nullable TemplateInfo templateInfo, @Nullable Map<String, String> importAliases)
+                                              String nameExpression, @Nullable TemplateInfo templateInfo, @Nullable Map<String, String> importAliases, String labelColor)
```
Comment (labkey-susanh) [inline experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]:
```suggestion
                                              String nameExpression, @Nullable TemplateInfo templateInfo, @Nullable Map<String, String> importAliases, @Nullable String labelColor)

```

**Review by labkey-susanh — APPROVED** [review]
Just suggestions here.

File: api/src/org/labkey/api/exp/api/SampleTypeService.java (hunk 1 above)
Comment (cnathe) [reply api/src/org/labkey/api/exp/api/SampleTypeService.java]: Done.

File: experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java (hunk 2 above)
Comment (cnathe) [reply experiment/src/org/labkey/experiment/api/SampleTypeServiceImpl.java]: Done

**Pull request merged**